### PR TITLE
feat: first attempt at helper function

### PR DESCRIPTION
**Pull request by danielgent**
As mentioned in another PR, it makes sense to extract out a helper function here

### CODE REVIEW

File: src/routes/tsx/routes.tsx [hunk 1]
```diff
@@ -28,6 +28,31 @@ import { validateCode } from "../../authentication-flows/passwordless";
...
+    throw new HTTPException(400, { message: "Tenant not found" });
+  }
+
+  const vendorSettings = await env.fetchVendorSettings(
+    session.authParams.client_id,
+  );
+
+  initI18n(tenant.language || "sv");
+
+  return { vendorSettings, client, tenant, session };
```
Comment (danielgent) [inline src/routes/tsx/routes.tsx:53]: I didn't overthink the name or what was happening here :smile:

File: src/routes/tsx/routes.tsx [hunk 2]
```diff
@@ -333,27 +302,9 @@ export const login = new OpenAPIHono<{ Bindings: Env }>()
...
-      }
-
-      const tenant = await env.data.tenants.get(client.tenant_id);
-      if (!tenant) {
-        throw new HTTPException(400, { message: "Tenant not found" });
-      }
-
-      initI18n(tenant.language || "sv");
-
-      const tenantNameInVendorStyles = tenant.name.toLowerCase();
```
Comment (danielgent) [inline src/routes/tsx/routes.tsx:353]: :thinking: I wonder why I was doing this, and only on this route :thinking: 

This is the one endpoint that is being tested on login2 as part of a full flow... so maybe now a test will fail

File: src/routes/tsx/routes.tsx [hunk 3]
```diff
@@ -574,12 +495,6 @@ export const login = new OpenAPIHono<{ Bindings: Env }>()
         console.log("User not found");
       }
 
-      const vendorSettings = await env.fetchVendorSettings(
-        session.authParams.client_id,
-      );
-
-      initI18n("sv");
```
Comment (danielgent) [inline src/routes/tsx/routes.tsx:581]: Ooops! Due to all those different PRs, I still had code in that was always set to `sv`, so doing this PR removed some potential bugs

**Review by markusahlstrand — APPROVED** [review]
It looks good. One option would be to do a openapi middleware and have it writing to the ctx.var, but.. this is clearer.

**Comment by danielgent** [comment]
> It looks good. One option would be to do a openapi middleware and have it writing to the ctx.var, but.. this is clearer.

There are no more routes either :smile:  (famous last words)

**Pull request merged**